services/class.tx_caretakerinstance_TYPO3VersionTestService.php

```diff
@@ -108,9 +108,13 @@ public function runTest()
      */
     protected function checkForLatestVersion($versionString, $allowUnstable = false)
     {
-        if (strpos($versionString, '.latest') !== false) {
+        if (strpos($versionString, '.latest') !== false || strpos($versionString, '.secure') !== false) {
             $versionDigits = explode('.', $versionString, 3);
-            $latestVersions = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('TYPO3\CMS\Core\Registry')->get('tx_caretaker', $allowUnstable ? 'TYPO3versions' : 'TYPO3versionsStable');
+            $versionSource = 'TYPO3versionsSecurity';
+            if (strpos($versionString, '.latest') !== false) {
+                $versionSource = $allowUnstable ? 'TYPO3versions' : 'TYPO3versionsStable';
+            }
+            $latestVersions = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('TYPO3\CMS\Core\Registry')->get('tx_caretaker', $versionSource);
             $newVersionString = $latestVersions[$versionDigits[0] . '.' . $versionDigits[1]];
```

Comment:
Should we cleanup this code as well? This might raise notices for undefined version keys.